quixstreams/kafka/admin.py

```diff
@@ -0,0 +1,191 @@
+        """
+        futures_dict = {}
+        cluster_topics = self.list_topics()
+        if existing_topics := [topic for topic in topics if topic in cluster_topics]:
+            futures_dict = self._admin_client.describe_configs(
+                [ConfigResource(2, topic) for topic in existing_topics]
+            )
+        configs = {
+            config_resource.name: {c.name: c.value for c in config.result().values()}
+            for config_resource, config in futures_dict.items()
+        }
+        return {
+            topic: TopicConfig(
+                num_partitions=len(cluster_topics[topic].partitions),
+                replication_factor=len(cluster_topics[topic].partitions[0].replicas),
+                extra_config=configs[topic],
+            )
+            if topic in existing_topics
+            else None
+            for topic in topics
+        }
+
+    def _finalize_create(self, futures: Dict[str, Future], timeout: int):
+        """
+        The confirmation step for topic creation.
+
+        :param futures: a dict of futures as generated by Confluent's
+            `AdminClient.create_topics()`
+        :param timeout: How long to confirm topics before raising exception
+        """
+        exceptions = {}
+        successful = []
+        existed = []
+        stop_time = time.time() + timeout
+        while futures and time.time() < stop_time:
+            time.sleep(1)
+            for topic_name in list(futures.keys()):
+                future = futures[topic_name]
+                if future.done():
+                    try:
+                        future.result()
+                        successful.append(topic_name)
+                    except KafkaException as e:
+                        # Topic was maybe created by another instance
+                        if e.args[0].name() == "TOPIC_ALREADY_EXISTS":
+                            existed.append(topic_name)
+                        else:
+                            exceptions[topic_name] = e.args[0].str()
+                    # Not sure how these get raised, but they are supposedly possible
+                    except (TypeError, ValueError) as e:
+                        exceptions[topic_name] = e
+                    del futures[topic_name]
+        if successful:
+            logger.info(f"Successfully created topics: {successful}")
+        if existed:
+            logger.info(f"These topics already exist: {existed}")
+        if exceptions:
+            raise self.CreateTopicFailure(exceptions)
+        if futures:
+            raise self.CreateTopicTimeout(
+                f"Timed out waiting for creation status for topics:\n"
+                f"{pprint.pformat([topic_name for topic_name in futures])}"
+            )
+
+    def create_topics(
+        self, topics: TopicList, timeout: int = 10, finalize_timeout: int = 20
+    ):
+        """
+        Create the given list of topics and confirm they are ready.
```

Comment:
not sure if `create_topics` is the best name here, because it can happen that some topics already exist and it will no create them.
Here are few of my suggestion, but feel free to use any name you think fits best:
`ensure_topics_created`, `setup_topics `, `initialize_topics`.

With this I would mofify the first sentence in the docs to for ex.:
   `Ensures that the specified Kafka topics are created and ready for use.`

Reply:
I agree...I thought similarly when making the function name. In the end, I didn't feel it was necessary to make a complicated name around the operation, as it usually had to be long to avoid running into similar ambiguity issues and made it kinda clunky to call.

I think as long as we are explicit in the docstring about the behavior (like you mentioned) + logging what it's doing (especially when topics already exist), its behavior should be pretty clear. I'll be sure to add those things. 

Do you think that's sufficient?